### DISCUSSION TITLE
SDXL and Sampler

### DISCUSSION THREAD

**Discussion by toyssamurai** [vladmandic/sdnext#2172] [Q&A]
My go-to sampler for pre-SDXL has always been DPM 2M. However, ever since I started using SDXL, I have found that the results of DPM 2M have become inferior. At approximately 25 to 30 steps, the results always appear as if the noise has not been completely resolved. Initially, I thought it was due to my LoRA model being overfitted. However, upon double-checking, I discovered that I faced the same problem even when using the base model. Even when I increased the number of steps to 99, I could still spot some very minor noise beneath the surface.

I have attached a comparison of using DPM 2M, DEIS, and the Euler sampler at 25, 50, 75, and 90 steps below. In the first comparison, I used it with my LoRA model. In the second one, I used just the base SDXL (at higher steps, you may need to zoom in for a closer look to identify the issue I am referring to). You can observe that when I used the Euler sampler, the image was already as smooth as silk at 25 steps. However, even when the base model is used, you can tell from the background that some noises aren't fully resolved when DEIS and DPM 2M are used. The issue is that Euler's smoothness has never been my favorite sampler for generating photographic images. I wonder if I did something wrong. I've tried toggling Karras on/off and experimented with all three DPM solver algorithms, but I haven't seen any significant improvement. That said, at very high steps, the mildly unresolved noisy effect kind of resembles subsurface scattering, which could be seen as beneficial for a portrait photo.

![a](https://github.com/vladmandic/automatic/assets/14955849/b173da58-69a5-4aa5-b860-cf88109f10fc)

![b](https://github.com/vladmandic/automatic/assets/14955849/eec12d19-1885-465d-8cfb-79c65448c385)

Is this the expected behavior? Did I do something wrong? Oh, below is my Sampler settings:

![samplers](https://github.com/vladmandic/automatic/assets/14955849/eed8cd15-5e3c-409a-bda1-68e09bf34bb4)

**Comment by toyssamurai** [comment]
One more thing. At 25 steps, you can see that the image is very noisy when DPM 2M is used. However, if I enable second pass, set the second pass sampler to Euler, I can "fix" the noise even when Refiner Start is set to an extremely low value of 0.99 and Refiner Step is set to just 1 (see the image below). 

![20230911-122719-231158-2561142425-1024x1024-s25-c6-d0 3-photo of ayakaimoto woman parted lips innocent expression-DPM 2M-sd_xl_base](https://github.com/vladmandic/automatic/assets/14955849/2d83579f-32a5-47c9-a0e9-51700498fead)

**Comment by vladmandic** [comment]
valid notes, but afaik its what it is - sdxl is just a different beast.

**Reply by toyssamurai** [reply]
That's all I need to know. I'm not sure if I did something wrong, as I keep hearing people recommending DEIS/DPM for SDXL.

I am just very surprised that such a small amount of refinement using Euler could make such a big difference in the final output.

**Reply by toyssamurai** [reply]
Just when I thought this issue was settled, I discovered that using the same sampler in Automatic1111 produces a completely different result. This prompted me to test all the DPM samplers in Automatic1111 to see if I could replicate the same problem. As you can see below, the same issue does not occur in Automatic1111 (Steps set at 25):

![auto1111](https://github.com/vladmandic/automatic/assets/14955849/885c1e43-058a-4766-bc7b-6f706ba34198)

At first, I thought that because I had set the DPM solver algorithm to sde-dpmsolver++ and had the samplers use Karras where applicable, I should be able to replicate the exact issue when I chose DPM++ 2M SDE Karras. However, not only did I not see the same noisy problem, but the generated image actually differed quite a bit from the one I generated using SD.Next (for example, the outfit the subject was wearing was entirely different). I got a similar outfit when using DPM++ 2M Karras, although it's not exactly the same. I did experience some noisy problems when using DPM++2M SDE and DPM++2M SDE Heun, but they don't look like the same kind of noise as the one in SD.Next's outputs.

While I don't know enough about how the backend works technically, I believe that there must still be something not quite right in SD.Next when it's generating images using the DPM sampler. It could be related to my settings, but I actually did a fresh reinstallation when I conducted my comparison yesterday, and I made very few changes to the settings. My feeling is that there may be some default settings that are not working well with the DPM samplers.

**Comment by monoinyo** [comment]
i'm all about uniPC in SDXL
